src/services/peerconnection.ts

```diff
@@ -250,8 +261,12 @@ export class PeerConnectionHelper {
 
             // Rebind close event
             dc.onclose = () => {
-                this.log.info(`Signalling data channel closed`);
-                link.closed();
+                if (this.closed) {
```

Comment:
You're using `this` here and `self` above... Should probably be `self` in both cases for explicitness (although `this` should work as well).